src/transpose.c

```diff
@@ -54,17 +59,10 @@ SEXP transpose(SEXP l, SEXP fill, SEXP ignoreArg, SEXP keepNamesArg) {
     const int len = length(li);
     if (ignore && len==0) continue;
     if (TYPEOF(li) != maxtype) {
-      li = PROTECT(isFactor(li) ? asCharacterFactor(li) : coerceVector(li, maxtype));
+      li = PROTECT(isFactor(li) ? (listCol ? coerceVector(asCharacterFactor(li), VECSXP) : asCharacterFactor(li)) : coerceVector(li, maxtype));
     } else PROTECT(li); // extra PROTECT just to help rchk by avoiding two counter variables
     switch (maxtype) {
-    case LGLSXP : {
```

Comment:
any reason for this? if we already have the `LGLSXP` branch written out I would keep it... "cheating" and collapsing the `LGLSXP` and `INTSXP` branches can be forgiven but shouldn't be encouraged IMO